src/repo/fs/blocks.rs

```diff
@@ -37,12 +37,26 @@ pub struct FsBlockStore {
     written_bytes: AtomicU64,
 }
 
+/// Helper is used to remove our key from `FsBlockStore::writes`. It is quite inefficient, some
+/// kind of reference counting would be great.
+///
+/// [`Drop`] is used to clean up the so that it is _safer_ to drop the future returned by
+/// `FsBlockStore::put`.
+///
+/// Without reference counting, there is a race condition with repeat multiple
+/// concurrent writers and dropping; this might lead into the first ones dropping the latter
+/// concurrent writes key.
```

Comment:
```suggestion
/// concurrent writes.
```

Reply:
I think I was going for this but deduplicated:

```suggestion
/// concurrent writes [`FsBlockStore::writes`] key.
```

As I was thinking of repeated "volleys" of writes:

```
for _ in 0..10 {
  let volley = async move {
    let mut stores = Vec::new();
    for _ in 0..10 {
      let fut = blocks.put(Block { cid: some_cid.clone(), data: todo!() });
      // spawn abortable
      // store abortable handle in stores
    }
    for handle in stores {
      handle.abort();
    }
  };
  tokio::spawn(volley);
}
```

The above would of course require fault injection to cause the blocking to fail the write. Perhaps this is quite unlikely situation.